src/bluetooth_interface.c

```diff
@@ -0,0 +1,165 @@
+ * "ERROR: Connection failed (error 0x%02x)". This function is called by the Zephyr
+ * Bluetooth stack. No need to call it manually.
+ */
+static void callback_ble_connect(struct bt_conn *connection, uint8_t error) {
+	if (error) {
+		LOG_ERR("Connection failed (error 0x%02x).", error);
+	} else {
+		LOG_INF("A new BLE connection handled!");
+	}
+}
+
+/* This function is called when the BLE connection is terminated. If the connection
+ * is terminated, the function will print "Disconnected (reason 0x%02x)". This function
+ * is called by the Zephyr Bluetooth stack. No need to call it manually.
+ */
+static void callback_ble_disconnect(struct bt_conn *connection, uint8_t reason) {
+	LOG_INF("BLE disconnected (reason 0x%02x).", reason);
+}
+
+/* This macro defines the connection callbacks. The callbacks are called by the Zephyr
+ * Bluetooth stack. No need to call them manually. Notice the similarity with the Linux
+ * kernel callback macros.
+ */
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = callback_ble_connect,
+	.disconnected = callback_ble_disconnect,
+};
+
+/* This function is the main handler for the BLE advertisement. It should be called
+ * after the BLE services are started. The function will start the BLE advertisement. 
+ */
+void start_bluetooth_advertisement(void) {
+	LOG_INF("Bluetooth initialized.");
+
+	int error = bt_le_adv_start(
+        BT_LE_ADV_CONN_NAME,
+        advertisment_data,
+        ARRAY_SIZE(advertisment_data),
+        NULL,
+        0
+    );
+
+	if (error) {
+		LOG_ERR("Advertising failed to start (err %d)", error);
+		return;
+	}
+
+	LOG_DBG("Advertising successfully started.");
+}
+
+/* This function is called when the BLE pairing is cancelled. If the pairing is
+ * cancelled, the function will print "Pairing cancelled: %s.". This function is
+ * called by the Zephyr Bluetooth stack. No need to call it manually.
+ */
+static void callback_ble_pairing_cancel(struct bt_conn *connection) {
+	char addr[BT_ADDR_LE_STR_LEN];
+	bt_addr_le_to_str(bt_conn_get_dst(connection), addr, sizeof(addr));
+
+	LOG_ERR("BLE pairing cancelled: %s.", addr);
+}
+
+/* This struct defines the BLE pairing callbacks. The callbacks are called by the
+ * Zephyr Bluetooth stack. No need to call them manually. Notice the similarity with
+ * the Linux kernel callback macros.
+ */
+struct bt_conn_auth_cb callback_ble_display_info = {
+	.cancel = callback_ble_pairing_cancel,
+};
+
+
+ssize_t read_u16(struct bt_conn *conn, const struct bt_gatt_attr *attr, void *buf, uint16_t len, uint16_t offset) {
+	const uint16_t *u16 = attr->user_data;
+	uint16_t value = sys_cpu_to_le16(*u16);
+ 	return bt_gatt_attr_read(conn, attr, buf, len, offset, &value, sizeof(value));
+}
+
+
+void notify_ble_connected_device(struct values_for_ble_t *value_to_notify) {
+	// Save the int part and dec part of the reading into an array.
+	temp_reading = value_to_notify->temp_reading_int * 100 + value_to_notify->temp_reading_dec * 0.0001;
```

Comment:
When there is some kind of conversion it is better to use function like macros for readbility and for elimating the duplication, such as, 

```suggestion
	temp_reading = SCALED_NUM_TO_INT(value_to_notify->temp_reading_int, value_to_notify->temp_reading_dec)
```